### ISSUE TITLE
[Set Audio Device] Unknown system error upon executing command

### ISSUE THREAD

**Issue by djpowers** [raycast/extensions#1660]
# Extension – [Set Audio Device](https://github.com/raycast/extensions/tree/9705ba94af72208cff790222ee3ce28e02231825/extensions/audio-device/)
Author: @benvp

## Description
All commands are failing for me with the following error:
```
There was an error fetching the audio devices. Command failed with Unknown system error -86: [~homepath]...
```
<img width="886" alt="Screen Shot 2022-05-10 at 11 05 06 PM" src="https://user-images.githubusercontent.com/4978418/167760744-de451b16-59a3-4d6f-80d8-8ca7b1f5101f.png">

## Steps To Reproduce

1. Install extension
2. Attempt to run either "Set Output Device" or "Set Input Device" commands

## The current behavior
Error message (pictured above)

## The expected behavior
Audio inputs/outputs should be set accordingly.

### Raycast version
Version 1.34.1

Of note, I am running macOS 12.3.1 on an M1 computer that currently does _not_ have Rosetta 2 installed. I had seen the "system error -86" crop up on another extension where a dependency appeared to require that Rosetta be present, so I suspect that could be the case here.

Assuming so, I'm not sure if there's a way to get this working without Rosetta being present, but that would be preferred if possible.

**Comment by djpowers** [comment]
Did a little bit more digging on this, and the full error reads as follows:
```
FAILURE There was an error fetching the audio devices. Command failed with Unknown system error -86: ~Library/Application Support/com.raycast.macos/extensions/audio-device/audio-devices list --output --json
```

It appears this is occurring upstream on the macos-audio-devices package, as I'm also getting an error trying to install it directly: `Bad CPU type in executable`

I opened up an issue on that package to see it's feasible to compile for M1 chips: karaggeorge/macos-audio-devices#23

**Comment by djpowers** [comment]
Bumping to keep open

**Comment by djpowers** [comment]
@pernielsentikaer, would it be possible to reopen this one? The issue is still occurring for me so I was hoping to keep it tracked. Thanks!

**Comment by djpowers** [comment]
Issue is still occurring.

**Comment by djpowers** [comment]
Still valid

**Comment by djpowers** [comment]
Still relevant

**Comment by djpowers** [comment]
Bump!

**Comment by pernielsentikaer** [comment]
@benvp do you have an idea here if we can do anything?

**Comment by merlindru** [comment]
+1

**Comment by benvp** [comment]
Never had this issue before. Do you have that checkbox checked? This determines which implementation for getting the audio devices is being used.

<img width="266" alt="CleanShot 2023-02-07 at 12 53 30@2x" src="https://user-images.githubusercontent.com/981344/217237968-5173c33d-684b-47ab-9cb8-3a4ac3209837.png">

**Comment by merlindru** [comment]
Thanks for replying! It wasn't checked. When I checked it, I didn't get the error anymore, but "No Results" instead @benvp

**Comment by benvp** [comment]
So it uses the `audio-devices` implementation. That's using a binary which lives in the repo. So probably this one is still an intel-compiled version and requires Rosetta. Would it be possible for you to check, if installing rosetta fixes it?

**Comment by merlindru** [comment]
@benvp Yes!!! It works now 😄 Thank you so much

For anyone coming from google, install Rosetta 2 by going to **Terminal > `softwareupdate --install-rosetta`**

**Comment by benvp** [comment]
You're welcome. So we should think about a way of adding an Apple Silicon specific binary.

**Comment by djpowers** [comment]
I noted in the original issue that this was likely an issue related to Rosetta 2 not being present, but agreed that it would be great to include Apple Silicon support.

The issue I opened upstream is linked above, and has since had a pull request opened to build a universal binary in karaggeorge/macos-audio-devices#27. I took a look at those changes and am hoping they'll be merged upstream, but if not we could potentially use a forked version with the same change.
